File: migrations/rename_user_name_to_username.js
import mongoose from "mongoose";
import dotenv from "dotenv";
import path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

dotenv.config({ path: path.resolve(__dirname, "../.env") });
dotenv.config();

const uri = process.env.MONGO_URI;

if (!uri) {
  console.error("MONGO_URI is not set. Add it to backend/.env or project root .env.");
  process.exit(1);
}

async function run() {
  await mongoose.connect(uri);
  const users = mongoose.connection.collection("users");

  const result = await users.updateMany(
    { name: { $exists: true } },
    [
      { $set: { username: { $ifNull: ["$username", "$name"] } } },
      { $unset: "name" }
    ]
  );

  console.log(
    `Rename complete. Matched: ${result.matchedCount}, Modified: ${result.modifiedCount}`
  );

  await mongoose.disconnect();
}

run().catch((err) => {
  console.error("Migration failed:", err);
  process.exit(1);
});

File: backend/migrations/rename_user_name_to_username.py
"""
One-time migration: rename 'name' field to 'username' in users collection.
Run from backend folder: python -m migrations.rename_user_name_to_username
"""
import asyncio
import os
from pathlib import Path

from dotenv import load_dotenv
from motor.motor_asyncio import AsyncIOMotorClient

_backend_dir = Path(__file__).resolve().parent.parent
load_dotenv(_backend_dir / ".env")
load_dotenv(_backend_dir.parent / ".env")
load_dotenv()

MONGO_URI = os.getenv("MONGO_URI")
if not MONGO_URI:
    print("MONGO_URI is not set. Add it to backend/.env or project root .env.")
    raise SystemExit(1)


async def run():
    client = AsyncIOMotorClient(MONGO_URI)
    db = client.get_default_database()
    users = db.users
    result = await users.update_many(
        {"name": {"$exists": True}},
        [
            {"$set": {"username": {"$ifNull": ["$username", "$name"]}}},
            {"$unset": "name"},
        ],
    )
    print(f"Rename complete. Matched: {result.matched_count}, Modified: {result.modified_count}")
    client.close()


if __name__ == "__main__":
    asyncio.run(run())
